### PR TITLE
fix(hygen): fixes hygen + eslint bug

### LINKED ISSUE #283: Components generated with hygen break pre commit hook
Committing a new component made with `yarn generate-component` fails with
```
error: Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
```

This is because hygen is creating a `story.js` file instead of a `story.ts` or `story.tsx` file.

I did a little [research](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/parser/README.md#parseroptionsproject) on it and`tsconfig.json#includes` could be updated to lint js files but I figure it's better to just make the story file in typescript.

### PR DESCRIPTION
**Pull request by mikemfleming**
resolves #283 

Generating a new component and committing it would fail on the pre commit lint step. This updates makes the following updates:

* Generating a new component creates a `ts` story file instead of `js`
* Excludes `example` and `src/stories` from estlint

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "root": true,
   "parser": "@typescript-eslint/parser",
   "parserOptions": {
-    project: "tsconfig.json"
```
Comment (pixelbandito) [inline .eslintrc:5]: WAT

**Review by pixelbandito — APPROVED** [review]
One super teeny thing

File: _templates/component/new/story.js.t [hunk 2]
```diff
@@ -1,11 +1,11 @@
 ---
-to: src/components/<%= h.changeCase.pascal(name) %>/story.js
+to: src/components/<%= h.changeCase.pascal(name) %>/story.tsx
```
[the remark this answers is not in the document]
Comment (pixelbandito) [reply _templates/component/new/story.js.t:2]: Yeah, that's a good call.
I think it'll work regardless, but that's much cleaner.

[the remark this answers is not in the document]
Comment (mikemfleming) [reply _templates/component/new/story.js.t:2]: The template file itself shouldn't have to match but I had the same thought. I'm also a hygen noob, but I was able to verify that this works as expected by runnint

```
yarn generate-component some-component

# make some changes

git add .
git commit -m "chore(hygen): testing that a newly generated component passes lint step"

# see that commit succeeds
```

**Pull request merged**